astroid/bases.py

```diff
@@ -436,7 +435,12 @@ def _infer_builtin_new(
         node_context = context.extra_context.get(caller.args[0])
         infer = caller.args[0].infer(context=node_context)
 
-        yield from (Instance(x) if x is not Uninferable else x for x in infer)  # type: ignore[misc,arg-type]
+        for inferred in infer:
```

Comment:
I can't suggest it, but can we skip the assignment on L436 and just iterate directly?